### PR TITLE
Fabio Mignogna - UI Design Patterns Challenge

### PR DESCRIPTION
**Pull request by migfabio**

### CODE REVIEW

File: MVVM Module/Tests/Feed UI/Helpers/FeedViewController+TestHelpers.swift [hunk 1]
```diff
@@ -54,6 +54,18 @@ extension FeedViewController {
...
+		return errorView?.message
+	}
+
+	func simulateTapOnErrorMessage() {
+		errorView?.button.simulateTap()
+	}
+
+	private var errorView: ErrorView? {
+		return tableView.tableHeaderView as? ErrorView
+	}
```
Comment (danillahtin) [inline MVVM Module/Tests/Feed UI/Helpers/FeedViewController+TestHelpers.swift:67]: Following on my previous comment this getter can be removed

```suggestion
```

File: MVP Module/Tests/Feed UI/FeedUISnapshotTests.swift [hunk 2]
```diff
@@ -51,6 +44,6 @@ class FeedUISnapshotTests: XCTestCase {
 
 private extension FeedViewController {
 	func display(errorMessage: String) {
-		fatalError("Must be implemented - follow the MVC solution as a guide")
+		errorView.show(message: errorMessage)
```
Comment (danillahtin) [inline MVP Module/Tests/Feed UI/FeedUISnapshotTests.swift:47]: If at some point error displaying logic will get more complex other than simply showing a message in errorView then this method would potentially break the test. To decouple the test I recommend using `FeedErrorViewModel` and `display` method of `FeedErrorView` protocol.
```suggestion
		display(FeedErrorViewModel(message: errorMessage))
```

File: MVVM Module/Tests/Feed UI/FeedUISnapshotTests.swift [hunk 3]
```diff
@@ -65,7 +58,7 @@ private class FeedLoaderStub: FeedLoader {
 
 private extension FeedViewController {
 	func display(errorMessage: String) {
-		fatalError("Must be implemented - follow the MVC solution as a guide")
+		errorView.show(message: errorMessage)
```
Comment (danillahtin) [inline MVVM Module/Tests/Feed UI/FeedUISnapshotTests.swift:61]: Following the same comment on MVP module, `viewModel.onErrorStateChange` can be used here to decouple the test from the way of displaying an error
```suggestion
		viewModel?.onErrorStateChange?(errorMessage)
```

**Review by danillahtin — COMMENTED** [review]
Hi @migfabio,

This is Danil, I am an instructor at Essential Developer.

Congratulations, you've passed the challenge 🥇

- All tests are passing ✅
- Your code is clean and handles edge cases ✅
- The UI and animations look and behave as expected ✅
- Light and Dark mode support ✅
- Added an extra language ✅

I left some review comments. Feel free to push changes.

Great job!

**Comment by migfabio** [comment]
Thanks for your review @danillahtin 
I've addressed all your feedback!

**Review by danillahtin — COMMENTED** [review]
Hi @migfabio!

Nice one!
